code/datums/extensions/demolisher/_demolisher.dm

```diff
@@ -0,0 +1,38 @@
+// A subtype for items that can dismantle/demolish non-reinforced non-natural walls.
+/datum/extension/demolisher
+	base_type = /datum/extension/demolisher
+	expected_type = /obj/item
+	var/demolish_delay = 6 SECONDS
+	var/demolish_verb
+	var/demolish_sound
+
+/datum/extension/demolisher/New(datum/holder, _delay, _verb, _sound)
+	. = ..()
+	if(_delay)
+		demolish_delay = _delay
+	if(_verb)
+		demolish_verb = _verb
+	if(_sound)
+		demolish_sound = _sound
+
+/datum/extension/demolisher/proc/get_demolish_delay(turf/wall/wall)
+	return demolish_delay - wall?.get_material()?.cut_delay
```

Comment:
Is it supposed to be - delay here? Wouldn't that make demolishing diamond wall instantenous?